octavia_proxy/api/drivers/elbv3/driver.py

```diff
@@ -113,10 +115,38 @@ def loadbalancer_create(self, session, loadbalancer):
         LOG.debug('Creating loadbalancer %s' % loadbalancer.to_dict())
 
         lb_attrs = loadbalancer.to_dict()
+        lb_attrs.pop('loadbalancer_id', None)
+
+        if 'pools' in lb_attrs:
+            lb_attrs.pop('pools')
+        if 'listeners' in lb_attrs:
+            lb_attrs.pop('listeners')
+        if 'vip_subnet_id' in lb_attrs:
+            lb_attrs['vip_subnet_cidr_id'] = lb_attrs['vip_subnet_id']
+        if 'vip_network_id' in lb_attrs:
+            lb_attrs['elb_virsubnet_ids'] = [lb_attrs.pop('vip_network_id')]
+        azs = lb_attrs.pop('availability_zone', 'eu-nl-01')
+        lb_attrs['availability_zone_list'] = azs.replace(' ', '').split(',')
+
         if 'tags' in lb_attrs:
             lb_attrs['tags'] = self._resource_tags(lb_attrs['tags'])
-        lb_attrs.pop('loadbalancer_id', None)
-        lb_attrs = elbv3_foremapping(lb_attrs)
+
+        if 'flavor_id' in lb_attrs:
+            flavors = []
+            if re.match(r's\d.\w+', lb_attrs['flavor_id']):
+                for item in ['L7', 'L4']:
+                    flavors.extend(session.vlb.find_flavor(
```

Comment:
ok, so here we have 2 calls to the API trying to get flavor with this name, fail, fetch list with filter resulting in overall 4 API calls. This is not really efficient